brilirs/src/interp.rs

```diff
@@ -201,12 +250,17 @@ impl fmt::Display for Value {
       Self::Float(v) => write!(f, "{v:.17}"),
       Self::Char(c) => write!(f, "{c}"),
       Self::Pointer(p) => write!(f, "{p:?}"),
+      Self::Atomic(a) => write!(f, "atomic_{a}"),
       Self::Uninitialized => unreachable!(),
     }
   }
 }
 
-fn optimized_val_output<T: std::io::Write>(out: &mut T, val: &Value) -> Result<(), std::io::Error> {
+fn optimized_val_output<T: std::io::Write>(
+  out: &Mutex<T>,
+  val: &Value,
+) -> Result<(), std::io::Error> {
+  let mut out = out.lock().unwrap();
```

Comment:
Maybe a nit, but there isn't really a reason to change this function right? Can we acquire the lock first and then call this function as a separation of concerns? That is equivalent to what you need to do if you are using Display instead.